[feat] 빌드 API , lastbuild stats

leewoojin12 opened the pull request:
---
name: "Pull Request"
about: "새로운 Pull Request 생성 시 사용합니다"
title: "[FEAT] Job 트리거 , lastbuildstatus"
labels: []
assignees: []
---



**1. 변경 요약 (Summary of Changes)**  
- Job 수동 트리거 기능 추가 (`/build/trigger/{jobName}`)
- 
- 각 Job의 마지막 빌드 상태(LASTBUILD STATUS) 확인 API 추가 (`/build/status/{jobName}`)

- Backend에 `BuildController`, `BuildService`, `BuildLog` 엔티티 및 관련 DTO 신규 추가

---

**2. 테스트 방법 (How to Test)**

- **로컬 환경에서 재현 방법:**
  1. 백엔드 서버 실행
  2. Swagger 또는 Postman으로 `/build/trigger/{jobName}` 호출 → Jenkins에서 해당 job 빌드 트리거 확인
  3. `/build/status/{jobName}` 호출 → last build 결과 (`SUCCESS`, `FAILURE`, 등) 확인

- **테스트 시나리오:**
  - **단위 테스트:**
  - **통합 테스트:**
    - API 호출 → Jenkins 연동 및 응답 검증
  - **e2e 테스트:**
    

**3. 관련 이슈 (Related Issues)**
- 연결된 이슈 번호: `#2`

**4. 체크리스트 (Checklist)**
- [x] 코드 컨벤션을 준수했는가? (파일명, 네이밍, 포맷 등)
- [x] 새로운 기능/버그 수정에 대한 단위 테스트 작성 및 통과했는가?
- [x] 통합 테스트 및 시나리오 테스트를 통과했는가?
- [x] 문서(Docs) 업데이트가 필요한 경우 반영했는가?
- [x] 주요 로직에 적절한 주석이 포함되었는가?
- [x] 보안/성능 고려 사항 검토했는가?
- [x] 관련 브랜치 전략 및 머지 대상이 적절한가?

leewoojin12 committed:
[feat] 빌드 API , lastbuild stats

chan0o0seo committed:
[docs] PR시 디스코드 알림

Pull request 생성되면 디스코드 채널로 알림오게 만들었습니다.

chan0o0seo committed:
Merge pull request #7 from baepo-minjok/feat/pr_notification

[DOC] PR시 디스코드 알림

leewoojin12 committed:
Merge remote-tracking branch 'refs/remotes/origin/develop' into feat/2-job_trigger

chan0o0seo commented on backend/src/main/java/com/example/backend/build/AppConfig.java:
```diff
@@ -0,0 +1,18 @@
+
+@Configuration
+
+public class AppConfig
```
이름이 너무 포괄적인거같아요. RestTemplate 관리하는 Config면 RestTemplateConfig가 적절한거같네요

chan0o0seo commented on backend/src/main/java/com/example/backend/build/BuildService.java:
```diff
@@ -0,0 +1,126 @@
+        String username = "admin";
+        String apiToken = "114cf329b1dbcc4e92dfca3c0d46f3c980";
+
+        // 1. Crumb 발급
```
User한테서 jenkins secret token을 받아서 사용하면 Crumb 발급이 필요없어요

chan0o0seo commented on backend/src/main/java/com/example/backend/build/BuildService.java:
```diff
@@ -0,0 +1,126 @@
+
+
+
+    public void triggerJenkinsBuild(String branch, String env) {
```
branch만 받는게 아니라 유저정보를 받아서 해당 유저의 jenkins 정보를 가져와서 트리거해야 할거같네요

chan0o0seo commented on backend/src/main/java/com/example/backend/build/BuildService.java:
```diff
@@ -0,0 +1,126 @@
+                triggerUrl, HttpMethod.POST, entity, String.class
+        );
+
+        System.out.println("Build trigger status: " + response.getStatusCode());
```
왠만하면 System.out보다는 log를 써주세여

chan0o0seo commented on backend/src/main/java/com/example/backend/build/BuildService.java:
```diff
@@ -0,0 +1,126 @@
+
+
+
+    public LastBuildStatusResponse getLastBuildStatus(String job) {
```
여기도 마찬가지로 job만 받으면 안될거같네요

chan0o0seo commented on backend/src/main/java/com/example/backend/build/LastBuildStatusResponse.java:
```diff
@@ -0,0 +1,79 @@
+@AllArgsConstructor
+@Builder
+
+public class LastBuildStatusResponse
```
코딩컨벤션한거보면 도메인+Response/Request+Dto 양식으로 하고 이너 클래스로 하기로 했는데 양식에 안 맞아여

leewoojin12 committed:
Merge remote-tracking branch 'refs/remotes/origin/main' into feat/2-job_trigger

leewoojin12 committed:
[fix] 빌드 API 요청 방식 및 파싱 로직 개선

- JobType(enum) 기반으로 latest/history 구분 처리
- RestTemplate 요청 공통화
- JSON 파싱 예외 처리 강화
- 로그 처리 추가 (info/error 구분)

leewoojin12 committed:
[fix] 트리거 빌드 로직 개선

- 특정 Trigger 발동
